src/core/cgroup.c

```diff
@@ -4175,9 +4175,23 @@ int unit_cgroup_freezer_action(Unit *u, FreezerAction action) {
         if (!cg_freezer_supported())
                 return 0;
 
+        /* Ignore all requests to freeze or thaw init.scope or -.slice */
+        if (unit_has_name(u, SPECIAL_ROOT_SLICE) || unit_has_name(u, SPECIAL_INIT_SCOPE))
+                return 0;
```

Comment:
should be `return action == FREEZE_FREEZE ? -EPERM : 0`, i.e. we want to refuse freezing explicitly of the root slice and init.scope